pkg/trustengine/trustengine.go

```diff
@@ -26,10 +26,11 @@ type Configuration struct {
 	Token     string
 }
 
+// GetToken requests a single token
 func GetToken(refName string, client *piperhttp.Client, trustEngineConfiguration Configuration) (string, error) {
 	secrets, err := GetSecrets([]string{refName}, client, trustEngineConfiguration)
 	if err != nil {
-		return "", err
+		return "", errors.Join(err, errors.New("couldn't get token from trust engine"))
```

Comment:
What is `errors.Join` doing? Would `errors.Wrap` be suitable as well?

Reply:
`Wrap` comes from [this package](https://github.com/pkg/errors) that isn't maintained anymore. I wanted to try out Golang's native [`errors.Join`](https://pkg.go.dev/errors#Join). Not sure how this'll look like in the logs and it's something I need to check out.

Reply:
As discussed, we use `errors.Wrap()` for now to stay consistent. But in the future, we should look into `errors.Join()` since we depend on an archived package for it.